crypto/finite_field.ts

```diff
@@ -195,5 +195,29 @@ function createField(p: bigint, t: bigint, twoadicRoot: bigint) {
       }
       return BigInt('0b' + binary.reverse().join(''));
     },
+    leftShift(x: bigint, bits: number) {
```

Comment:
Unit tests in `o1js` were failing when `leftShift` was defined like this:

```ts
leftShift(x: bigint, bits: number) {
    return mod(x << BigInt(bits), p);
}
```

I'm wondering if the reviewer knows why this is the case? I had to define a manual shift implementation instead to get unit tests to pass.

Reply:
Its probably because the shift operator in JavaScript (on `bigint`) does not make a "cut" after 64 bit and just keeps shifting. I believe bitwise operations on `number` are limited to 31bit, `bigint` isn't

Reply:
```ts
leftShift(1n, 64)
> 0
1n << 65n
> 36893488147419103232
```

Reply:
Fixed in https://github.com/o1-labs/o1js-bindings/pull/191/commits/169c97fa2ab1ef1ee345496795a067756fd8eab2